tests/ui/cmp_owned.stderr

```diff
@@ -43,10 +43,16 @@ error: this creates an owned instance just for comparison
    |     ^^^^^^^^^^^^^^^^^^ help: try: `y == x`
 
 error: this creates an owned instance just for comparison
-  --> $DIR/cmp_owned.rs:49:9
+  --> $DIR/cmp_owned.rs:47:5
    |
-49 |         self.to_owned() == *other
+47 |     y.to_owned() == **x;
+   |     ^^^^^^^^^^^^^^^^^^^ help: try: `y == x`
```

Comment:
will that suggestion still compile? I think with another level of indirection it won't. I'm not sure how to correctly do this. Maybe trim dereferences until the types match up?

Reply:
Ah, you are correct. The simple thing to do here is to just remove the first deref on the right hand side, because you are removing a single `to_owned` on the left. I just pushed an implementation of that.

Reply:
If you start out with `&&&&&T` on the left, then the `to_owned` will deref all of them except for the last one. Now, that won't work anymore. We also don't know how many derefs on the `x` are really necessary, since some might go throught a `Deref` impl that doesn't just yield another reference, but completely changes the type (think `String` dereferencing to `str`).